diff -urNp kmix-4.9.0.orig/core/mixer.cpp kmix-4.9.0/core/mixer.cpp
--- kmix-4.9.0.orig/core/mixer.cpp	2012-07-11 10:18:52.000000000 +0200
+++ kmix-4.9.0/core/mixer.cpp	2012-08-02 20:52:30.000000000 +0200
@@ -38,8 +38,8 @@
 QList<Mixer *> Mixer::s_mixers;
 MasterControl Mixer::_globalMasterCurrent;
 MasterControl Mixer::_globalMasterPreferred;
-float Mixer::VOLUME_STEP_DIVISOR = 20;
-float Mixer::VOLUME_PAGESTEP_DIVISOR = 10;
+float Mixer::VOLUME_STEP_DIVISOR = 100;
+float Mixer::VOLUME_PAGESTEP_DIVISOR = 100;
 bool Mixer::m_beepOnVolumeChange = false;
 
 int Mixer::numDrivers()
