--- src/r8101_n.c.orig  2013-07-04 13:17:40.049985664 +0200
+++ src/r8101_n.c   2013-07-04 13:24:43.158974654 +0200
@@ -1906,7 +1906,11 @@
     }
 #else
     if (opts2 & RxVlanTag)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)
         __vlan_hwaccel_put_tag(skb, swab16(opts2 & 0xffff));
+#else
+        __vlan_hwaccel_put_tag(skb, htons(ETH_P_8021Q), swab16(opts2 & 0xffff));
+#endif
 #endif
 
     desc->opts2 = 0;
@@ -1967,7 +1971,11 @@
     else
         tp->cp_cmd &= ~RxChkSum;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)
     if (dev->features & NETIF_F_HW_VLAN_RX)
+#else
+    if (dev->features & NETIF_F_HW_VLAN_CTAG_RX)
+#endif
         tp->cp_cmd |= RxVlan;
     else
         tp->cp_cmd &= ~RxVlan;
@@ -7292,7 +7300,11 @@
 #endif
 
 #ifdef CONFIG_R8101_VLAN
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)
     dev->features |= NETIF_F_HW_VLAN_TX | NETIF_F_HW_VLAN_RX;
+#else
+    dev->features |= NETIF_F_HW_VLAN_CTAG_TX | NETIF_F_HW_VLAN_CTAG_RX;
+#endif
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)
     dev->vlan_rx_kill_vid = rtl8101_vlan_rx_kill_vid;
 #endif//LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)
@@ -7305,7 +7317,11 @@
 #else
     dev->features |= NETIF_F_RXCSUM | NETIF_F_SG;
     dev->hw_features = NETIF_F_SG | NETIF_F_IP_CSUM | NETIF_F_TSO |
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)
                        NETIF_F_RXCSUM | NETIF_F_HW_VLAN_TX | NETIF_F_HW_VLAN_RX;
+#else
+                       NETIF_F_RXCSUM | NETIF_F_HW_VLAN_CTAG_TX | NETIF_F_HW_VLAN_CTAG_RX;
+#endif
     dev->vlan_features = NETIF_F_SG | NETIF_F_IP_CSUM | NETIF_F_TSO |
                          NETIF_F_HIGHDMA;
 #endif
