diff -Naur r8101-1.029.00.orig/src/r8101_n.c r8101-1.029.00/src/r8101_n.c
--- r8101-1.029.00.orig/src/r8101_n.c	2016-07-31 01:13:28.364226433 +0200
+++ r8101-1.029.00/src/r8101_n.c	2016-07-31 01:14:21.559229007 +0200
@@ -10209,7 +10209,6 @@
                         dev->hw_features &= ~NETIF_F_IPV6_CSUM;
                         netif_set_gso_max_size(dev, LSO_32K);
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,18,0)
-                        dev->gso_min_segs = NIC_MIN_PHYS_BUF_COUNT;
                         dev->gso_max_segs = NIC_MAX_PHYS_BUF_COUNT_LSO_64K;
 #endif
                 } else {
@@ -10217,7 +10216,6 @@
                         dev->features |=  NETIF_F_IPV6_CSUM | NETIF_F_TSO6;
                         netif_set_gso_max_size(dev, LSO_64K);
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,18,0)
-                        dev->gso_min_segs = NIC_MIN_PHYS_BUF_COUNT;
                         dev->gso_max_segs = NIC_MAX_PHYS_BUF_COUNT_LSO2;
 #endif
                 }
