diff -Naur firefox-47.0b1.orig/js/src/old-configure.in firefox-47.0b1/js/src/old-configure.in
--- firefox-47.0b1.orig/js/src/old-configure.in	2016-04-26 17:07:56.000000000 +0200
+++ firefox-47.0b1/js/src/old-configure.in	2016-05-01 10:36:15.502323075 +0200
@@ -1172,14 +1172,6 @@
                   AC_MSG_RESULT([no])
                   LDFLAGS=$_SAVE_LDFLAGS)
 
-    AC_MSG_CHECKING([for --build-id option to ld])
-    _SAVE_LDFLAGS=$LDFLAGS
-    LDFLAGS="$LDFLAGS -Wl,--build-id"
-    AC_TRY_LINK(,,AC_MSG_RESULT([yes])
-                  [NSPR_LDFLAGS="$NSPR_LDFLAGS -Wl,--build-id"],
-                  AC_MSG_RESULT([no])
-                  LDFLAGS=$_SAVE_LDFLAGS)
-
     MOZ_SET_WARNINGS_CFLAGS
 
     _DEFINES_CFLAGS='-include $(topobjdir)/js/src/js-confdefs.h -DMOZILLA_CLIENT'
diff -Naur firefox-47.0b1.orig/old-configure.in firefox-47.0b1/old-configure.in
--- firefox-47.0b1.orig/old-configure.in	2016-04-26 17:07:52.000000000 +0200
+++ firefox-47.0b1/old-configure.in	2016-05-01 10:37:10.098325717 +0200
@@ -1366,14 +1366,6 @@
                   AC_MSG_RESULT([no])
                   LDFLAGS=$_SAVE_LDFLAGS)
 
-    AC_MSG_CHECKING([for --build-id option to ld])
-    _SAVE_LDFLAGS=$LDFLAGS
-    LDFLAGS="$LDFLAGS -Wl,--build-id"
-    AC_TRY_LINK(,,AC_MSG_RESULT([yes])
-                  [NSPR_LDFLAGS="$NSPR_LDFLAGS -Wl,--build-id"],
-                  AC_MSG_RESULT([no])
-                  LDFLAGS=$_SAVE_LDFLAGS)
-
     AC_MSG_CHECKING([for --ignore-unresolved-symbol option to ld])
     HAVE_LINKER_SUPPORT_IGNORE_UNRESOLVED=
     _SAVE_LDFLAGS=$LDFLAGS
